contributors/design-proposals/service_links.md

```diff
@@ -0,0 +1,237 @@
+# Explicit service links
+
+## Abstract
+
+Proposes an API for explicit links to services in the pod spec.
+
+## Motivation
+
+Currently, all pods are automatically injected with docker-link-style
+environment variables for each service in their namespaces.  Making service
+dependencies explicit in the PodSpec has a number of advantages:
+
+1.  Reduce accidental coupling
+2.  Reduce the number of IP tables rules that have to be created by the
+    kube-proxy
+3.  Identify relationships between pods and services, allowing user interfaces
+    to display additional information for end users.
```

Comment:
This is more a replacement for the default service environment variables.

ConfigMap isn't really usable for providing information about service ports and IPs, since some controller would need to populate those fields based on data from the Service API.

For loose coupling, we want something like #254, which is a building block for service binding as part of the service catalog effort.